openid-4-verifiable-presentations-1_0.md

```diff
@@ -542,6 +542,266 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+
+`vct_values`:
+: OPTIONAL. An array of strings that specifies allowed values for
+the type of the requested Verifiable Credential. All elements in the array MUST
+be valid type identifiers as defined in [@!I-D.ietf-oauth-sd-jwt-vc]. The Wallet
+MAY return credentials that inherit from any of the specified types, following
+the inheritance logic defined in [@!I-D.ietf-oauth-sd-jwt-vc].
+
+`credential_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies
+the allowed algorithms for signing the SD-JWT (i.e., only the issuer-signed part).
+
+`kbjwt_signing_alg_values`:
+: OPTIONAL. An array of strings that specifies the
+allowed algorithms for signing the Key-Binding JWT (KB-JWT) (i.e., the part signed
+by the holder).
+
+`sd_alg_values`:
+: OPTIONAL. An array of strings that specifies the allowed hash algorithms for
+digests in the SD-JWT and KB-JWT.
+
+### Format `mso_mdoc` {#format_mso_mdoc}
+
+TBD
+
+### Format `jwt_vp*`
+
+TBD
+
+## Claim Query {#claim_query}
+
+A claim query is an object that describes one or more claims that are to be
+returned. Depending on the format of the requested Verifiable Credential, the
+claim query can contain different properties.
+
+
+
+### JSON-based Claims Structures {#json_based_claims}
+
+For credentials containing JSON-based claims structures, the following
+properties are defined:
+
+`path`:
+: REQUIRED. A claims path pointer that specifies the path to the claim
+within the Verifiable Credential, as defined in (#claims_path_pointer).
+
+### ISO 18013-5 Credentials {#iso_18013-5}
+
+For ISO 18013-5 Credentials, the following properties are defined:
+
+`namespace`:
+: REQUIRED. A string that specifies the namespace of the claim
+within the Verifiable Credential, e.g., `org.iso.18013.5.1`.
+
+`claim_name`:
+: REQUIRED. A string that specifies the name of the claim within the provided namespace
+in the Verifiable Credential, e.g., `first_name`.
+
+### Claims Path Pointer {#claims_path_pointer}
+
+A claims path pointer is a pointer into the JSON structure of the Verifiable
+Credential, identifying one or more claims. A claims path pointer MUST be a
+non-empty array of strings and non-negative integers. A `*` string value indicates
+that all elements of the currently selected array(s) are to be selected; any other
+string indicates that the respective key is to be selected; and a non-negative
+integer indicates that the respective index in an array is to be selected. The path
+is formed as follows:
+
+  - Start with an empty array.
+  - To address a particular claim within an object, append the key (claim name)
+    to the array.
+  - To address an element within an array, append the index to the array (as a
+    non-negative, 0-based integer).
+  - To address all elements within an array, append the string `*` to the array.
+
+Verifiers MUST NOT point to the same claim more than once in a single query.
+Wallets SHOULD ignore such duplicate claim queries.
```

Comment:
I think it very likely that verifiers will accidentally point to the same claim more than once, likely when mixing claim indexes with claim names, and certainly when mixing the wildcard with either or both claim indexes and/or claim names.

It might be viable to forbid use of the wildcard in combination with integers and/or strings, but that is not sufficient to cover all scenarios.

```suggestion
Verifiers SHOULD NOT point to the same claim more than once in a single query,
e.g., by combining the wildcard `*` with one or more claim names or indexes,
or by using claim names *and* indexes.
Wallets SHOULD return each claim once, regardless of duplicate claim queries.
```

Reply:
Hi @TallTed I've hidden this comment because as I replied to a previous comment a week or two ago we don't appear to have a contribution agreement in place for you so can't use your contributions. If you'd like you contribute you can complete the contribution agreement here: https://openid.net/intellectual-property/openid-foundation-contribution-agreements/

If you are already covered by a contribution agreement, please let me know the details so we can locate it. Thank you!